### PR TITLE
Add deleteAsync and clearBlocking method for improving performance

### PR DESCRIPTION
**Pull request by itosho**
About this issue: https://github.com/cakephp/cakephp/issues/16360
Continuation from: https://github.com/cakephp/cakephp/pull/16478

### CODE REVIEW

**Comment by itosho** [comment]
@markstory @ADmad Please review 🙏

**Comment by dereuromark** [comment]
Should target 4.next

File: src/Cache/Engine/RedisEngine.php [hunk 1]
```diff
@@ -256,6 +271,37 @@ public function clear(): bool
         return $isAllDeleted;
     }
 
+    /**
+     * Blocking delete all keys from the cache
```
Comment (dereuromark) [inline src/Cache/Engine/RedisEngine.php:275]: Sounds weird, there might be a word Clear or alike missing at the beginning.

**Comment by ADmad** [comment]
Changing the target to 4.next pulls in unrelated commits, so you need to rebase your changes on top of 4.next HEAD.

**Comment by itosho** [comment]
I made a mistake and the history became complicated, so I will create a new branch and cherry-pick it.

**Pull request closed without merging**